server/src/category/category.controller.js

```diff
@@ -0,0 +1,21 @@
+import { NOT_FOUND, OK } from 'http-status';
+
+import { Category } from '@/shared/database/index';
+import { DatabaseError } from 'sequelize';
+import errorMessages from '@/shared/constants/errorMessages';
+import HttpError from '@/shared/error/httpError';
+
+const list = async (req, res, next) => {
+  try {
+    const categories = await Category.findAll();
+    return res.status(OK).json(categories);
+  } catch (err) {
+    if (err instanceof DatabaseError) {
+      return next(new HttpError(NOT_FOUND, errorMessages.NOT_FOUND_ERROR));
```

Comment:
not found will never be the case, that is only for when we add `where` or something like that. If you don't have anything in the db findAll will return an empty array, not an error. You can remove this if and just leave the `return next(new Error())`